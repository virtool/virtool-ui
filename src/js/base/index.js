export { Accordion } from "./accordion/Accordion";
export { AccordionContent } from "./accordion/AccordionContent";
export { AccordionTrigger } from "./accordion/AccordionTrigger";
export { ScrollingAccordionItem } from "./accordion/ScrollingAccordionItem";
export { Alert } from "./Alert";
export { AlertInner } from "./AlertInner";
export { AlertOuter } from "./AlertOuter";
export { Attribution } from "./Attribution";
export { AttributionWithName } from "./AttributionWithName";
export { Badge } from "./Badge";
export { Box } from "./Box";
export { BoxGroup } from "./BoxGroup";
export { BoxGroupDisabled } from "./BoxGroupDisabled";
export { BoxGroupHeader } from "./BoxGroupHeader";
export { BoxGroupSearch } from "./BoxGroupSearch";
export { BoxGroupSection } from "./BoxGroupSection";
export { BoxGroupTable } from "./BoxGroupTable";
export { BoxSpaced } from "./BoxSpaced";
export { BoxTitle } from "./BoxTitle";
export { device } from "./Breakpoints";
export { Button } from "./Button";
export { Checkbox } from "./Checkbox";
export { CheckboxLabel } from "./CheckboxLabel";
export { Circle } from "./Circle";
export { CloseButton } from "./CloseButton";
export { Color } from "./Color";
export { ComboBox } from "./combobox/ComboBox";
export { Container } from "./Container";
export { ContainerNarrow } from "./ContainerNarrow";
export { ContainerSide } from "./ContainerSide";
export { ContainerWide } from "./ContainerWide";
export { Dialog } from "./Dialog";
export { DialogAlert } from "./DialogAlert";
export { DialogContent } from "./DialogContent";
export { DialogContext } from "./DialogContext";
export { DialogFooter } from "./DialogFooter";
export { DialogOverlay } from "./DialogOverlay";
export { DialogTabs } from "./DialogTabs";
export { DialogTitle } from "./DialogTitle";
export { DialogTrigger } from "./DialogTrigger";
export { DividerHorizontal } from "./DividerHorizontal";
export { DividerVertical } from "./DividerVertical";
export { Dropdown } from "./Dropdown";
export { DropdownButton } from "./DropdownButton";
export { DropdownMenuContent } from "./DropdownMenuContent";
export { DropdownMenuDownload } from "./DropdownMenuDownload";
export { DropdownMenuItem } from "./DropdownMenuItem";
export { DropdownMenuLink } from "./DropdownMenuLink";
export { ExternalLink } from "./ExternalLink";
export { Icon } from "./Icon";
export { IconButton } from "./IconButton";
export { InitialIcon } from "./InitialIcon";
export { Input } from "./Input";
export { InputContainer } from "./InputContainer";
export { InputContext } from "./InputContext";
export { InputError } from "./InputError";
export { InputGroup } from "./InputGroup";
export { InputIcon } from "./InputIcon";
export { InputIconButton } from "./InputIconButton";
export { InputLabel } from "./InputLabel";
export { InputLevel } from "./InputLevel";
export { InputLoading } from "./InputLoading";
export { InputPassword } from "./InputPassword";
export { InputSearch } from "./InputSearch";
export { InputSelect } from "./InputSelect";
export { InputSimple } from "./InputSimple";
export { Key } from "./Key";
export { Label } from "./Label";
export { LinkButton } from "./LinkButton";
export { Loader } from "./Loader";
export { LoadingPlaceholder } from "./LoadingPlaceholder";
export { Logo } from "./Logo";
export { Markdown } from "./Markdown";
export { NoneFound } from "./NoneFound";
export { NoneFoundBox } from "./NoneFoundBox";
export { NoneFoundSection } from "./NoneFoundSection";
export { NotFound } from "./NotFound";
export { Pagination } from "./pagination/Pagination";
export { ProgressBar } from "./ProgressBar";
export { ProgressBarAffixed } from "./ProgressBarAffixed";
export { ProgressCircle } from "./ProgressCircle";
export { RelativeTime } from "./RelativeTime";
export { RemoveBanner } from "./RemoveBanner";
export { SaveButton } from "./SaveButton";
export { SectionHeader } from "./SectionHeader";
export { Select } from "./select/Select";
export { SelectButton } from "./select/SelectButton";
export { SelectContent } from "./select/SelectContent";
export { SelectItem } from "./select/SelectItem";
export { SelectBox } from "./SelectBox";
export { SelectBoxGroupSection } from "./SelectBoxGroupSection";
export { SidebarHeader } from "./SidebarHeader";
export { SidebarHeaderButton } from "./SidebarHeaderButton";
export { SideBarSection } from "./SideBarSection";
export { SlashList } from "./SlashList";
export { SubviewHeader } from "./SubviewHeader";
export { SubviewHeaderAttribution } from "./SubviewHeaderAttribution";
export { SubviewHeaderTitle } from "./SubviewHeaderTitle";
export { Table } from "./Table";
export { Tabs } from "./Tabs";
export { TabsLink } from "./TabsLink";
export { TextArea } from "./TextArea";
export { Toolbar } from "./Toolbar";
export { ViewHeader } from "./ViewHeader";
export { ViewHeaderAttribution } from "./ViewHeaderAttribution";
export { ViewHeaderIcons } from "./ViewHeaderIcons";
export { ViewHeaderSubtitle } from "./ViewHeaderSubtitle";
export { ViewHeaderTitle } from "./ViewHeaderTitle";
