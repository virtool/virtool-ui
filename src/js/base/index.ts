import InputGroup from "@base/InputGroup.js";
import InputLabel from "@base/InputLabel.js";

export { InputGroup, InputLabel };

export { Link } from "wouter";
export { Checkbox } from "./Checkbox";
export { CloseButton } from "./CloseButton";
export { Color } from "./Color";
export { ComboBox } from "./ComboBox";
export { Container } from "./Container";
export { ContainerNarrow } from "./ContainerNarrow";
export { ContainerSide } from "./ContainerSide";
export { ContainerWide } from "./ContainerWide";
export { Dialog } from "./Dialog";
export { DialogAlert } from "./DialogAlert";
export { DialogContent } from "./DialogContent";
export { DialogContext } from "./DialogContext";
export { DialogFooter } from "./DialogFooter";
export { DialogOverlay } from "./DialogOverlay";
export { DialogTabs } from "./DialogTabs";
export { DialogTitle } from "./DialogTitle";
export { DialogTrigger } from "./DialogTrigger";
export { Dropdown } from "./Dropdown";
export { DropdownButton } from "./DropdownButton";
export { DropdownMenuContent } from "./DropdownMenuContent";
export { DropdownMenuDownload } from "./DropdownMenuDownload";
export { DropdownMenuItem } from "./DropdownMenuItem";
export { DropdownMenuLink } from "./DropdownMenuLink";
export { ExternalLink } from "./ExternalLink";
export { Icon } from "./Icon";
export { IconButton } from "./IconButton";
export { InitialIcon } from "./InitialIcon";
export { Input } from "./Input";
export { InputContainer } from "./InputContainer";
export { InputContext } from "./InputContext";
export { InputError } from "./InputError";
export { InputIcon } from "./InputIcon";
export { InputIconButton } from "./InputIconButton";
export { InputLoading } from "./InputLoading";
export { InputPassword } from "./InputPassword";
export { InputSearch } from "./InputSearch";
export { InputSelect } from "./InputSelect";
export { InputSimple } from "./InputSimple";
export { Key } from "./Key";
export { Label } from "./Label";
export { LinkButton } from "./LinkButton";
export { Loader } from "./Loader";
export { LoadingPlaceholder } from "./LoadingPlaceholder";
export { Logo } from "./Logo";
export { Markdown } from "./Markdown";
export { NoneFound } from "./NoneFound";
export { NoneFoundBox } from "./NoneFoundBox";
export { NoneFoundSection } from "./NoneFoundSection";
export { NotFound } from "./NotFound";
export { Pagination } from "./Pagination";
export { ProgressBar } from "./ProgressBar";
export { ProgressBarAffixed } from "./ProgressBarAffixed";
export { ProgressCircle } from "./ProgressCircle";
export { RelativeTime } from "./RelativeTime";
export { RemoveBanner } from "./RemoveBanner";
export { SaveButton } from "./SaveButton";
export { SectionHeader } from "./SectionHeader";
export { Select } from "./Select";
export { SelectBox } from "./SelectBox";
export { SelectBoxGroupSection } from "./SelectBoxGroupSection";
export { SelectButton } from "./SelectButton";
export { SelectContent } from "./SelectContent";
export { SelectItem } from "./SelectItem";
export { SidebarHeader } from "./SidebarHeader";
export { SidebarHeaderButton } from "./SidebarHeaderButton";
export { SideBarSection } from "./SideBarSection";
export { SlashList } from "./SlashList";
export { SubviewHeader } from "./SubviewHeader";
export { SubviewHeaderAttribution } from "./SubviewHeaderAttribution";
export { SubviewHeaderTitle } from "./SubviewHeaderTitle";
export { Table } from "./Table";
export { Tabs } from "./Tabs";
export { TabsLink } from "./TabsLink";
export { TextArea } from "./TextArea";
export { ViewHeader } from "./ViewHeader";
export { ViewHeaderAttribution } from "./ViewHeaderAttribution";
export { ViewHeaderIcons } from "./ViewHeaderIcons";
export { ViewHeaderSubtitle } from "./ViewHeaderSubtitle";
export { ViewHeaderTitle } from "./ViewHeaderTitle";
