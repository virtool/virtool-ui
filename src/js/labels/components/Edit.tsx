import { DialogPortal, DialogTrigger } from "@radix-ui/react-dialog";
import { useMutation, useQueryClient } from "@tanstack/react-query";
import React, { useState } from "react";
import { Request } from "../../app/request";
import { Dialog, DialogContent, DialogOverlay, DialogTitle, Icon } from "../../base";
import { StyledButtonSmall } from "../../base/styled/StyledButtonSmall";
import { labelQueryKeys } from "../queries";
import { LabelForm } from "./Form";

interface EditLabelProps {
    id: string;
    color: string;
    description: string;
    name: string;
}

export function EditLabel({ id, color, name, description }: EditLabelProps) {
    const [show, setShow] = useState(false);

    const queryClient = useQueryClient();

    const mutation = useMutation(data => {
        return Request.patch(`/labels/${id}`)
            .send(data)
            .then(response => {
                return response.body;
            });
    });

    const handleSubmit = data => {
        mutation.mutate(data, {
            onSuccess: () => {
                setShow(false);
                queryClient.invalidateQueries(labelQueryKeys.lists());
            },
        });
    };

    return (
        <Dialog open={show} onOpenChange={show => setShow(show)}>
            <StyledButtonSmall as={DialogTrigger}>
                <Icon name="pen" />
                <span>Edit</span>
            </StyledButtonSmall>
            <DialogPortal>
                <DialogOverlay />
                <DialogContent>
                    <DialogTitle>Edit a label</DialogTitle>
                    <LabelForm color={color} description={description} name={name} onSubmit={handleSubmit} />
                </DialogContent>
            </DialogPortal>
        </Dialog>
    );
}
