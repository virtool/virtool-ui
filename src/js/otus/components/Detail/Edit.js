import { get } from "lodash-es";
import React from "react";
import { connect } from "react-redux";
import { Modal, ModalHeader } from "../../../base";
import { clearError } from "../../../errors/actions";
import { getTargetChange } from "../../../utils/utils";
import { editOTU, hideOTUModal } from "../../actions";
import { OTUForm } from "../Form";

const getInitialState = ({ name = "", abbreviation = "" }) => ({
    name,
    abbreviation,
    error: ""
});

class Edit extends React.Component {
    constructor(props) {
        super(props);
        this.state = getInitialState(props);
    }

    handleChange = e => {
        const { name, value, error } = getTargetChange(e.target);

        this.setState({
            [name]: value,
            [error]: ""
        });

        if (this.props.error) {
            this.props.onClearError();
        }
    };

    handleModalEnter = () => {
        this.setState(getInitialState(this.props));
    };

    handleModalExited = () => {
        if (this.props.error) {
            this.props.onClearError();
        }
    };

    handleSubmit = e => {
        e.preventDefault();

        if (!this.state.name) {
            return this.setState({
                errorName: "Required Field"
            });
        }

        if (!this.state.error) {
            this.props.onSave(this.props.otuId, this.state.name, this.state.abbreviation);
        }
    };

    render() {
        const error = this.state.error || this.props.error || "";

        return (
            <Modal
                label="Edit OTU"
                show={this.props.show}
                onEnter={this.handleModalEnter}
                onExited={this.handleModalExited}
                onHide={this.props.onHide}
            >
                <ModalHeader>Edit OTU</ModalHeader>
                <OTUForm
                    name={this.state.name}
                    abbreviation={this.state.abbreviation}
                    error={error}
                    onSubmit={this.handleSubmit}
                    onChange={this.handleChange}
                />
            </Modal>
        );
    }
}

const mapStateToProps = state => ({
    show: state.otus.edit,
    error: get(state, "errors.EDIT_OTU_ERROR.message", "")
});

const mapDispatchToProps = dispatch => ({
    onHide: () => {
        dispatch(hideOTUModal());
    },

    onSave: (otuId, name, abbreviation) => {
        dispatch(editOTU(otuId, name, abbreviation));
    },

    onClearError: () => {
        dispatch(clearError("EDIT_OTU_ERROR"));
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(Edit);
