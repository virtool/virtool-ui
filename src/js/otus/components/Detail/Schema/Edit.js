import { find, findIndex } from "lodash-es";
import PropTypes from "prop-types";
import React from "react";
import { connect } from "react-redux";
import { Button, Modal, ModalBody, ModalFooter, ModalHeader } from "../../../../base";
import SegmentForm from "./Form";

const getInitialState = props => ({
    newEntry: {
        name: props.curSeg.name,
        molecule: props.curSeg.molecule,
        required: props.curSeg.required,
        showError: false,
        nameTaken: false
    }
});

class EditSegment extends React.Component {
    constructor(props) {
        super(props);
        this.state = getInitialState(this.props);
    }

    handleChange = entry => {
        this.setState({
            newEntry: {
                name: entry.name,
                molecule: entry.molecule,
                required: entry.required
            }
        });
    };

    handleModalEnter = () => {
        this.setState(getInitialState(this.props));
    };

    handleSubmit = e => {
        e.preventDefault();

        const takenName = find(this.props.schema, ["name", this.state.newEntry.name]);

        if (takenName && takenName.name !== this.props.curSeg.name) {
            this.setState({
                newEntry: { ...this.state.newEntry, showError: false, nameTaken: true }
            });
        } else if (this.state.newEntry.name) {
            const newArray = this.props.schema.slice();
            const name = this.props.curSeg.name;
            const index = findIndex(newArray, ["name", name]);

            newArray[index] = this.state.newEntry;

            this.setState({
                newEntry: { ...this.state.newEntry, showError: false, nameTaken: false }
            });

            this.props.onSubmit(newArray);
        } else {
            this.setState({
                newEntry: { ...this.state.newEntry, showError: true, nameTaken: false }
            });
        }
    };

    render() {
        return (
            <Modal
                label="Edit Segment"
                show={this.props.show}
                onHide={this.props.onHide}
                onEnter={this.handleModalEnter}
            >
                <ModalHeader>Edit Segment</ModalHeader>
                <form onSubmit={this.handleSubmit}>
                    <ModalBody>
                        <SegmentForm onChange={this.handleChange} newEntry={this.state.newEntry} />
                    </ModalBody>
                    <ModalFooter>
                        <Button color="blue" icon="save" type="submit">
                            Save
                        </Button>
                    </ModalFooter>
                </form>
            </Modal>
        );
    }
}

EditSegment.propTypes = {
    schema: PropTypes.arrayOf(PropTypes.object),
    show: PropTypes.bool.isRequired,
    onHide: PropTypes.func,
    onSubmit: PropTypes.func,
    curSeg: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
    schema: state.otus.detail.schema
});

export default connect(mapStateToProps)(EditSegment);
