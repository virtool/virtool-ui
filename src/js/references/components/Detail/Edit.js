import React from "react";
import { connect } from "react-redux";
import { pushState } from "../../../app/actions";
import { Modal, ModalBody, ModalFooter, ModalHeader, SaveButton } from "../../../base";
import { getTargetChange, routerLocationHasState } from "../../../utils/utils";
import { editReference } from "../../actions";
import { ReferenceForm } from "../Form";

const getInitialState = detail => ({
    name: detail.name,
    description: detail.description,
    organism: detail.organism,
    errorName: ""
});

export class EditReference extends React.Component {
    constructor(props) {
        super(props);
        this.state = {};
    }

    handleChange = e => {
        const { name, value, error } = getTargetChange(e.target);

        if (name !== "name") {
            return this.setState({ [name]: value });
        }

        this.setState({
            [name]: value,
            [error]: ""
        });
    };

    handleModalEnter = () => {
        this.setState(getInitialState(this.props.detail));
    };

    handleSubmit = e => {
        e.preventDefault();

        if (!this.state.name.length) {
            this.setState({ errorName: "Required Field" });
        }

        if (this.state.name.length) {
            const { errorName, ...update } = this.state;
            this.props.onSubmit(this.props.detail.id, update);
            this.props.onHide();
        }
    };

    render() {
        return (
            <Modal label="Edit" show={this.props.show} onHide={this.props.onHide} onEnter={this.handleModalEnter}>
                <ModalHeader>Edit Reference</ModalHeader>
                <form onSubmit={this.handleSubmit}>
                    <ModalBody>
                        <ReferenceForm
                            description={this.state.description}
                            organism={this.state.organism}
                            mode={this.state.mode}
                            name={this.state.name}
                            errorName={this.state.errorSelect}
                            onChange={this.handleChange}
                        />
                    </ModalBody>

                    <ModalFooter>
                        <SaveButton />
                    </ModalFooter>
                </form>
            </Modal>
        );
    }
}

const mapStateToProps = state => ({
    show: routerLocationHasState(state, "editReference"),
    detail: state.references.detail
});

const mapDispatchToProps = dispatch => ({
    onSubmit: (refId, update) => {
        dispatch(editReference(refId, update));
    },

    onHide: () => {
        dispatch(pushState({ editReference: false }));
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(EditReference);
