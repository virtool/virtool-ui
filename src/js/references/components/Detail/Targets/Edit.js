import { find, get, map, toNumber } from "lodash-es";
import React from "react";

import { connect } from "react-redux";
import { pushState } from "../../../../app/actions";
import { Button, ModalBody, ModalFooter, Modal, ModalHeader } from "../../../../base";
import { routerLocationHasState } from "../../../../utils/utils";
import { editReference } from "../../../actions";
import { TargetForm } from "./Form";

const getInitialState = ({ name, description, length, required }) => ({
    name: name || "",
    description: description || "",
    length: length || 0,
    required: required || false,
    errorName: ""
});

export class EditTarget extends React.Component {
    constructor(props) {
        super(props);
        this.state = getInitialState(this.props);
    }

    handleSubmit = e => {
        e.preventDefault();

        if (!this.state.name) {
            return this.setState({
                errorName: "Required field"
            });
        }

        const targets = map(this.props.targets, target => {
            if (this.props.name === target.name) {
                return {
                    ...target,
                    name: this.state.name,
                    description: this.state.description,
                    length: toNumber(this.state.length),
                    required: this.state.required
                };
            }

            return target;
        });

        this.props.onSubmit(this.props.refId, { targets });
        this.props.onHide();
    };

    handleChange = e => {
        this.setState({
            [e.target.name]: e.target.value,
            required: e.target.checked
        });
    };

    handleEnter = () => {
        this.setState(getInitialState(this.props));
    };

    render() {
        return (
            <Modal show={this.props.show} onEnter={this.handleEnter} onHide={this.props.onHide} label="Edit Target">
                <ModalHeader>Edit Target</ModalHeader>
                <form onSubmit={this.handleSubmit}>
                    <ModalBody>
                        <TargetForm
                            onChange={this.handleChange}
                            name={this.state.name}
                            description={this.state.description}
                            length={this.state.length}
                            required={this.state.required}
                            errorName={this.state.errorName}
                        />
                    </ModalBody>

                    <ModalFooter>
                        <Button type="submit" icon="save" color="blue">
                            Submit
                        </Button>
                    </ModalFooter>
                </form>
            </Modal>
        );
    }
}

export const mapStateToProps = state => {
    const activeName = get(state, "router.location.state.editTarget") || "";
    let target = {};

    if (activeName) {
        target = find(state.references.detail.targets, { name: activeName }) || {};
    }

    const { name, description, length, required } = target;

    return {
        name,
        description,
        length,
        required,
        targets: state.references.detail.targets,
        refId: state.references.detail.id,
        show: routerLocationHasState(state, "editTarget")
    };
};

export const mapDispatchToProps = dispatch => ({
    onSubmit: (refId, update) => {
        dispatch(editReference(refId, update));
    },

    onHide: () => {
        dispatch(pushState({ editTarget: "" }));
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(EditTarget);
