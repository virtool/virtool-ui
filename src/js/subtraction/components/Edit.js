import React from "react";
import { connect } from "react-redux";
import { ModalBody, ModalFooter, Input, InputGroup, InputLabel, Modal, SaveButton, ModalHeader } from "../../base";
import { editSubtraction } from "../actions";

export class EditSubtraction extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            error: "",
            name: props.name,
            nickname: props.nickname
        };
    }

    handleChange = e => {
        const { name, value } = e.target;
        this.setState({
            [name]: value
        });
    };

    handleSubmit = e => {
        e.preventDefault();

        if (this.state.name === "") {
            return setState({
                error: "A name must be provided"
            });
        }

        this.props.onUpdate(this.props.id, this.state.name, this.state.nickname);
        this.props.onHide();
    };

    render() {
        return (
            <Modal label="Edit Subtraction" show={this.props.show} onHide={this.props.onHide}>
                <ModalHeader>Edit Subtraction</ModalHeader>
                <form onSubmit={this.handleSubmit}>
                    <ModalBody>
                        <InputGroup>
                            <InputLabel>Name</InputLabel>
                            <Input name="name" value={this.state.name} onChange={this.handleChange} />
                        </InputGroup>
                        <InputGroup>
                            <InputLabel>Nickname</InputLabel>
                            <Input name="nickname" value={this.state.nickname} onChange={this.handleChange} />
                        </InputGroup>
                    </ModalBody>

                    <ModalFooter>
                        <SaveButton />
                    </ModalFooter>
                </form>
            </Modal>
        );
    }
}

export const mapStateToProps = state => {
    const { id, name, nickname } = state.subtraction.detail;
    return { id, name, nickname };
};

export const mapDispatchToProps = dispatch => ({
    onUpdate: (id, name, nickname) => {
        dispatch(editSubtraction(id, name, nickname));
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(EditSubtraction);
