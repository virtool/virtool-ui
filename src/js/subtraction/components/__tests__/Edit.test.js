import { Input, Modal } from "../../../base";
import { EditSubtraction, mapDispatchToProps } from "../Edit";
describe("<EditSubtraction />", () => {
    let e;
    let props;

    beforeEach(() => {
        e = {
            preventDefault: jest.fn(),
            target: {
                value: "Foo"
            }
        };

        props = {
            id: "foo",
            name: "Prunus persica",
            nickname: "Peach",
            show: true,
            onHide: jest.fn(),
            onUpdate: jest.fn()
        };
    });

    it("should render", () => {
        const wrapper = shallow(<EditSubtraction {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render when [show=false]", () => {
        props.show = false;
        const wrapper = shallow(<EditSubtraction {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render after name is changed", () => {
        const wrapper = shallow(<EditSubtraction {...props} />);
        expect(wrapper).toMatchSnapshot();
        e.target.name = "name";
        wrapper.find(Input).at(0).simulate("change", e);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render after nickname is changed", () => {
        const wrapper = shallow(<EditSubtraction {...props} />);
        expect(wrapper).toMatchSnapshot();
        e.target.name = "nickname";
        wrapper.find(Input).at(1).simulate("change", e);
        expect(wrapper).toMatchSnapshot();
    });

    it("should call onUpdate() when form is submitted", () => {
        const wrapper = shallow(<EditSubtraction {...props} />);
        wrapper.find("form").simulate("submit", e);
        expect(props.onUpdate).toHaveBeenCalledWith("foo", "Prunus persica", "Peach");
        expect(props.onHide).toHaveBeenCalled();
    });

    it("should call onHide() when closed", () => {
        const wrapper = shallow(<EditSubtraction {...props} />);
        wrapper.find(Modal).prop("onHide")();
        expect(props.onHide).toHaveBeenCalledWith();
    });
});

describe("mapDispatchToProps()", () => {
    it("should return updateSubtraction in props", () => {
        const dispatch = jest.fn();
        const props = mapDispatchToProps(dispatch);
        props.onUpdate("foo", "Foo", "Bar");
        expect(dispatch).toHaveBeenCalledWith({
            subtractionId: "foo",
            name: "Foo",
            nickname: "Bar",
            type: "UPDATE_SUBTRACTION_REQUESTED"
        });
    });
});
