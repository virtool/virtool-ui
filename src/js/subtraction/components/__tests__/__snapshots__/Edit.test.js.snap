// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditSubtraction /> should render 1`] = `
<Modal
  label="Edit Subtraction"
  onHide={[MockFunction]}
  show={true}
>
  <ModalHeader>
    Edit Subtraction
  </ModalHeader>
  <form
    onSubmit={[Function]}
  >
    <Modal__ModalBody>
      <InputGroup>
        <Input__InputLabel>
          Name
        </Input__InputLabel>
        <Input
          name="name"
          onChange={[Function]}
          value="Prunus persica"
        />
      </InputGroup>
      <InputGroup>
        <Input__InputLabel>
          Nickname
        </Input__InputLabel>
        <Input
          name="nickname"
          onChange={[Function]}
          value="Peach"
        />
      </InputGroup>
    </Modal__ModalBody>
    <Modal__ModalFooter>
      <SaveButton />
    </Modal__ModalFooter>
  </form>
</Modal>
`;

exports[`<EditSubtraction /> should render after name is changed 1`] = `
<Modal
  label="Edit Subtraction"
  onHide={[MockFunction]}
  show={true}
>
  <ModalHeader>
    Edit Subtraction
  </ModalHeader>
  <form
    onSubmit={[Function]}
  >
    <Modal__ModalBody>
      <InputGroup>
        <Input__InputLabel>
          Name
        </Input__InputLabel>
        <Input
          name="name"
          onChange={[Function]}
          value="Prunus persica"
        />
      </InputGroup>
      <InputGroup>
        <Input__InputLabel>
          Nickname
        </Input__InputLabel>
        <Input
          name="nickname"
          onChange={[Function]}
          value="Peach"
        />
      </InputGroup>
    </Modal__ModalBody>
    <Modal__ModalFooter>
      <SaveButton />
    </Modal__ModalFooter>
  </form>
</Modal>
`;

exports[`<EditSubtraction /> should render after name is changed 2`] = `
<Modal
  label="Edit Subtraction"
  onHide={[MockFunction]}
  show={true}
>
  <ModalHeader>
    Edit Subtraction
  </ModalHeader>
  <form
    onSubmit={[Function]}
  >
    <Modal__ModalBody>
      <InputGroup>
        <Input__InputLabel>
          Name
        </Input__InputLabel>
        <Input
          name="name"
          onChange={[Function]}
          value="Foo"
        />
      </InputGroup>
      <InputGroup>
        <Input__InputLabel>
          Nickname
        </Input__InputLabel>
        <Input
          name="nickname"
          onChange={[Function]}
          value="Peach"
        />
      </InputGroup>
    </Modal__ModalBody>
    <Modal__ModalFooter>
      <SaveButton />
    </Modal__ModalFooter>
  </form>
</Modal>
`;

exports[`<EditSubtraction /> should render after nickname is changed 1`] = `
<Modal
  label="Edit Subtraction"
  onHide={[MockFunction]}
  show={true}
>
  <ModalHeader>
    Edit Subtraction
  </ModalHeader>
  <form
    onSubmit={[Function]}
  >
    <Modal__ModalBody>
      <InputGroup>
        <Input__InputLabel>
          Name
        </Input__InputLabel>
        <Input
          name="name"
          onChange={[Function]}
          value="Prunus persica"
        />
      </InputGroup>
      <InputGroup>
        <Input__InputLabel>
          Nickname
        </Input__InputLabel>
        <Input
          name="nickname"
          onChange={[Function]}
          value="Peach"
        />
      </InputGroup>
    </Modal__ModalBody>
    <Modal__ModalFooter>
      <SaveButton />
    </Modal__ModalFooter>
  </form>
</Modal>
`;

exports[`<EditSubtraction /> should render after nickname is changed 2`] = `
<Modal
  label="Edit Subtraction"
  onHide={[MockFunction]}
  show={true}
>
  <ModalHeader>
    Edit Subtraction
  </ModalHeader>
  <form
    onSubmit={[Function]}
  >
    <Modal__ModalBody>
      <InputGroup>
        <Input__InputLabel>
          Name
        </Input__InputLabel>
        <Input
          name="name"
          onChange={[Function]}
          value="Prunus persica"
        />
      </InputGroup>
      <InputGroup>
        <Input__InputLabel>
          Nickname
        </Input__InputLabel>
        <Input
          name="nickname"
          onChange={[Function]}
          value="Foo"
        />
      </InputGroup>
    </Modal__ModalBody>
    <Modal__ModalFooter>
      <SaveButton />
    </Modal__ModalFooter>
  </form>
</Modal>
`;

exports[`<EditSubtraction /> should render when [show=false] 1`] = `
<Modal
  label="Edit Subtraction"
  onHide={[MockFunction]}
  show={false}
>
  <ModalHeader>
    Edit Subtraction
  </ModalHeader>
  <form
    onSubmit={[Function]}
  >
    <Modal__ModalBody>
      <InputGroup>
        <Input__InputLabel>
          Name
        </Input__InputLabel>
        <Input
          name="name"
          onChange={[Function]}
          value="Prunus persica"
        />
      </InputGroup>
      <InputGroup>
        <Input__InputLabel>
          Nickname
        </Input__InputLabel>
        <Input
          name="nickname"
          onChange={[Function]}
          value="Peach"
        />
      </InputGroup>
    </Modal__ModalBody>
    <Modal__ModalFooter>
      <SaveButton />
    </Modal__ModalFooter>
  </form>
</Modal>
`;
